### DISCUSSION TITLE
Exclude routes from Vercel Speed Insights

### DISCUSSION THREAD

**Discussion by Griffin-Foster** [vercel/community#3391] [Help] [answered]
### Summary

I'm trying to figure out of there's a way to exclude specific routes from vercel's speed insights feature. On my website, I use Sanity to allow for my client to manage site content; however, whenever the Sanity Studio route is accessed it completely tanks the Speed Insights. However, the data from loading that page isn't relevant to the performance of the website because the Sanity CMS route isn't customer-facing.

Any help would be appreciated! :)

### Steps to Reproduce

No steps necessary. All routes are included in vercel's Speed Insights by default after enabling the feature.

**Comment by VegtMedia-Alex** [comment]
We have the same issue, would love a solution for it.

**Comment by SamKomesarook** [comment]
Upping this. We have multiple pages that are simple redirects for SEO, but they make our speed scores difficult to interpret. It would be amazing if we could exclude these.

**Comment by tomelliot** [comment]
I'm also trying to work out how to exclude pages that are part of the app from the speed insights, and only include pages that are on the marketing site.

**Comment by samuelhorn** [comment]
We have the same, an internal page that are used in an app are polluting the scores for the rest of the marketing site.
Vercel said that they're not supporting this right now, but that they're working on a solution that would make it possible.

I would appreciate having this feature as soon as possible. 👍🏼

**Comment by feugy** [comment]
Hello @Griffin-Foster, @samuelhorn, @tomelliot, @SamKomesarook  and @DB-Alex.

Yesterday we've released `@vercel/speed-insights` [package](https://vercel.com/docs/speed-insights/migrating-from-legacy#new-package:-@vercel/speed-insights), our new component for collecting your site's performance.

One of its benefits is the ability for you to specify a `beforeSend()` [function](https://vercel.com/docs/speed-insights/package#beforesend), so you could filter out this undesired traffic. 

Please try it and let us know if it fits your purpose.

**Reply by samuelhorn** [reply]
Very glad to hear this! Thanks!

I installed the package, and have it successfully running in my rootLayout as per the docs:
```
<SpeedInsights debug />
```

But when I try to intercept routes like:
```
<SpeedInsights
    debug
    beforeSend={(data) => {
        if (data.url.includes("/slice-simulator")) {
            return null;
        }
        return data;
    }}
/>
```

I get an error on the front-end:
```
Error: Functions cannot be passed directly to Client Components unless you explicitly expose it by marking it with "use server".
  <... debug={true} beforeSend={function}>
```

**Reply by feugy** [reply, marked as answer]
Indeed, this component is a client component at the moment, so you'll need to put it into a [file marked](https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs) with `'use client'`.

A setup along these lines should work:
```jsx
// layout.jsx
import { ClientScript } from './client-script'

export default function RootLayout({ children }) {
  return (
    <html lang="en">
      <body>
        {children}
        <ClientScript />
      </body>
    </html>
  )
}
```

```jsx
// client-script.jsx
'use client'
import { SpeedInsights } from '@vercel/speed-insights'

export function ClientScript() {
  return <SpeedInsights beforeSend={(data) => {...} />
}
```

**Reply by samuelhorn** [reply]
Ah, off course. Thanks!
